idea/src/org/jetbrains/kotlin/idea/update/GooglePluginUpdateVerifier.kt

```diff
@@ -32,16 +36,16 @@ class GooglePluginUpdateVerifier : PluginUpdateVerifier() {
             val pluginCompatibility = unmarshaller.unmarshal(stream) as PluginCompatibility
 
             val release = getRelease(pluginCompatibility) ?: return PluginVerifyResult.decline("no verified versions for this build")
-            val version = pluginDescriptor.version
-            if (release.plugins().any { version == it.version }) {
-                return PluginVerifyResult.accept()
-            } else {
-                return PluginVerifyResult.decline("version to be verified")
-            }
+            return if (release.plugins().any { KOTLIN_PLUGIN_ID == it.id && version == it.version }) PluginVerifyResult.accept()
+                   else PluginVerifyResult.decline("version to be verified")
         } catch (e : Exception) {
-            val reason = String.format("Failed to verify plugin %s version %s due to exception %s",
-                    pluginDescriptor.name, pluginDescriptor.version, e.message)
-            return PluginVerifyResult.decline(reason)
+            LOG.error("Exception when verifying plugin ${pluginDescriptor.pluginId.idString} version $version", e)
+            return when (e) {
+                is MalformedURLException, is IOException ->
```

Comment:
I think some minor fixes are still possible for exception handling.

For example, separate processing for `MalformedURLException` looks to be unneeded, because only a static URL is processed here.

What's more important, is that `IOException` probably shouldn't have `LOG.error()` call, because it will be possible to send such messages through exception reporter. These errors can be caused by user local network settings and nothing can be done about them by authors of the code.